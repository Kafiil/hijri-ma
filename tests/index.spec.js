const { getCurrentDate, getCurrentMonth, getAll } = require('../src');
const { localTime } = require('../src/time');
const { DATEFORMAT } = require('../src/config');
const arDate = '‪1441‬ ‫ذو القعدة‬ ‪1‬';
const date = "1 Thul-Qi'dah 1441";

jest.mock('../src/api');
jest.mock('../src/time');

//FIXME: correct the test name
describe('The main functionnalities', () => {
  it.each([
    [, arDate],
    [true, arDate],
    [false, date],
  ])('Result in %s should be %s', async (lang, expectedDate) => {
    const result = await getCurrentDate(lang);
    expect(result).toBe(expectedDate);
  });

  it('Should get the whole current month data', async () => {
    let day = localTime().format(DATEFORMAT);
    const result = await getCurrentMonth();
    expect(result).toMatchSnapshot();
    expect(Array.isArray(result)).toBe(true);
    expect(result.length).toBe(29);
    for (let i = 0; i < 29; i++) {
      expect(result[i].day).toBe(i + 1);
      expect(result[i].month).toBe(11);
      expect(result[i].year).toBe(1441);
      expect(result[i].gregorianDate).toBe(day);
      day = localTime(day).add(1, 'day').format(DATEFORMAT);
    }
  });
});
