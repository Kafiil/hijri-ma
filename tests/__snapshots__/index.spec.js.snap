// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The main functionnalities Should get the whole current month data 1`] = `
Array [
  Object {
    "day": 1,
    "georgianDate": "2020-06-23",
    "month": 11,
    "year": 1441,
  },
  Object {
    "day": 2,
    "georgianDate": "2020-06-24",
    "month": 11,
    "year": 1441,
  },
  Object {
    "day": 3,
    "georgianDate": "2020-06-25",
    "month": 11,
    "year": 1441,
  },
  Object {
    "day": 4,
    "georgianDate": "2020-06-26",
    "month": 11,
    "year": 1441,
  },
  Object {
    "day": 5,
    "georgianDate": "2020-06-27",
    "month": 11,
    "year": 1441,
  },
  Object {
    "day": 6,
    "georgianDate": "2020-06-28",
    "month": 11,
    "year": 1441,
  },
  Object {
    "day": 7,
    "georgianDate": "2020-06-29",
    "month": 11,
    "year": 1441,
  },
  Object {
    "day": 8,
    "georgianDate": "2020-06-30",
    "month": 11,
    "year": 1441,
  },
  Object {
    "day": 9,
    "georgianDate": "2020-07-01",
    "month": 11,
    "year": 1441,
  },
  Object {
    "day": 10,
    "georgianDate": "2020-07-02",
    "month": 11,
    "year": 1441,
  },
  Object {
    "day": 11,
    "georgianDate": "2020-07-03",
    "month": 11,
    "year": 1441,
  },
  Object {
    "day": 12,
    "georgianDate": "2020-07-04",
    "month": 11,
    "year": 1441,
  },
  Object {
    "day": 13,
    "georgianDate": "2020-07-05",
    "month": 11,
    "year": 1441,
  },
  Object {
    "day": 14,
    "georgianDate": "2020-07-06",
    "month": 11,
    "year": 1441,
  },
  Object {
    "day": 15,
    "georgianDate": "2020-07-07",
    "month": 11,
    "year": 1441,
  },
  Object {
    "day": 16,
    "georgianDate": "2020-07-08",
    "month": 11,
    "year": 1441,
  },
  Object {
    "day": 17,
    "georgianDate": "2020-07-09",
    "month": 11,
    "year": 1441,
  },
  Object {
    "day": 18,
    "georgianDate": "2020-07-10",
    "month": 11,
    "year": 1441,
  },
  Object {
    "day": 19,
    "georgianDate": "2020-07-11",
    "month": 11,
    "year": 1441,
  },
  Object {
    "day": 20,
    "georgianDate": "2020-07-12",
    "month": 11,
    "year": 1441,
  },
  Object {
    "day": 21,
    "georgianDate": "2020-07-13",
    "month": 11,
    "year": 1441,
  },
  Object {
    "day": 22,
    "georgianDate": "2020-07-14",
    "month": 11,
    "year": 1441,
  },
  Object {
    "day": 23,
    "georgianDate": "2020-07-15",
    "month": 11,
    "year": 1441,
  },
  Object {
    "day": 24,
    "georgianDate": "2020-07-16",
    "month": 11,
    "year": 1441,
  },
  Object {
    "day": 25,
    "georgianDate": "2020-07-17",
    "month": 11,
    "year": 1441,
  },
  Object {
    "day": 26,
    "georgianDate": "2020-07-18",
    "month": 11,
    "year": 1441,
  },
  Object {
    "day": 27,
    "georgianDate": "2020-07-19",
    "month": 11,
    "year": 1441,
  },
  Object {
    "day": 28,
    "georgianDate": "2020-07-20",
    "month": 11,
    "year": 1441,
  },
  Object {
    "day": 29,
    "georgianDate": "2020-07-21",
    "month": 11,
    "year": 1441,
  },
]
`;
